File: zhuldyz/leetcode/JanuaryChallenge/check-array-formation-through-concatenation.cpp
class Solution {
public:
    bool canFormArray(vector<int>& arr, vector<vector<int>>& pieces) {
        int ans[101] = {0};
        for (int i = 0; i < arr.size(); i++) {
            ans[arr[i]] = i+1;
        }
        for (int j = 0; j < pieces.size(); j++) {
            int count = ans[pieces[j][0]] -1;
            for (int i = 0; i <pieces[j].size(); i++) {
                if(ans[pieces[j][i]] == 0) return false;
                if(count == ans[pieces[j][i]] - 1) count = ans[pieces[j][i]];
                else return false;
            }
        }
        return true;
    }
};
//accepted

File: amina/leetcode/Check Array Formation Through Concatenation.java
class Solution {
    public boolean canFormArray(int[] arr, int[][] pieces) {
           HashMap<Integer, Integer> map = new HashMap<>();
            for(int i = 0; i< arr.length; i++){
                map.put(arr[i], i);
            }
        int prev;
            for( int i = 0; i < pieces.length; i++){
                for(int j = 0; j < pieces[i].length; j++){
                    if(!map.containsKey(pieces[i][j])){
                        return false;
                    }else{
                        if(j == 0){
                           prev = map.get(pieces[i][j]); 
                            continue;
                        }else{
                            prev = map.get(pieces[i][j-1]); 
                        }
                        int cur = map.get(pieces[i][j]);
                       if(cur < prev || prev+1!=cur){
                           return false;
                           
                       }
                    }
                }
            }
        return true;

    }
}
